backend/src/main/java/com/example/backend/Services/ReservationDisplayService.java

```diff
@@ -0,0 +1,81 @@
+package com.example.backend.Services;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+
+import com.example.backend.DTOMappers.PageResponseMapper;
+import com.example.backend.DTOMappers.ReservationMapper;
+import com.example.backend.DTOs.PageResponse;
+import com.example.backend.DTOs.ReservationDTO;
+import com.example.backend.DTOs.ReservationFilterCriteria;
+import com.example.backend.Entities.Reservation;
+import com.example.backend.Repositories.ReservationRepository;
+import com.example.backend.Specifications.ReservationSpecifications;
+import com.example.backend.Utilites.ValidateInput;
+
+@Service
+public class ReservationDisplayService {
+    
+    private final ReservationRepository reservationRepository;
+
+    public ReservationDisplayService (ReservationRepository reservationRepository){
+        this.reservationRepository = reservationRepository;
+    }
+
+    public PageResponse<ReservationDTO> getReservationByUserId (Integer userId, int pageNumber) {
+        ValidateInput.validateId(userId);
+        ValidateInput.validatePageNumber(pageNumber);
+
+        Pageable pageable = PageRequest.of(pageNumber, 10);
+        Page<Reservation> page = reservationRepository.findByUserId(userId, pageable);
+        Page<ReservationDTO> pageDTO = page.map(ReservationMapper :: toDTO);
+        return PageResponseMapper.toPageResponse(pageDTO);
+    }
+
+    public PageResponse<ReservationDTO> filterReserved (ReservationFilterCriteria filterDTO, int pageNumber) {
```

Comment:
could you check the front-end developer because we maybe don't need to have two functions just one and let the filter variables optional like what we ended to do the last milestone.

Reply:
There was many duplicated code as you mentioned, so I combined them into functions without much duplicates.

Reply:
I checked with the front-end and there was unneeded methods, so I removed them